ejb3/src/main/java/org/jboss/as/ejb3/timerservice/schedule/CalendarBasedTimeout.java

```diff
@@ -284,12 +284,16 @@ private Calendar computeNextTime(Calendar nextCal) {
             // no change in time
             return nextCal;
         }
+
+        // Set the time before adding the a day. If we do it after,
+        // we could be using an invalid DST value in setTime method
+        setTime(nextCal, nextHour, nextMinute, nextSecond);
+
```

Comment:
It seems unreal that we're not using `java.time` yet here.  But this looks OK to me.